lib/linux.js

```diff
@@ -159,5 +173,13 @@ exports.set = function set(imagePath) {
 
 	params[params.length - 1] = params[params.length - 1].replace('%s', path.resolve(imagePath));
 
+	if (app.cmd === 'gsettings') {
+		isCinnamon().then(cinnamon => {
```

Comment:
This won't work. This check is async, but you're assigning to the outside scope which will be done by the time this is executed.